Zoltan: new build warning

## Bug Report
I'm seeing a few warnings like the following:

```
[96/538] Building C object packages/zoltan/src/CMakeFiles/zoltan.dir/Utilities/Communication/comm_default.c.o
/ascldap/users/jhu/trilinos/Trilinos-trilinos-for-sake/packages/zoltan/src/Utilities/Communication/comm_default.c:62:25: warning: 'zoltan_global_mpi_lock' is static but used in inline function 'zoltan_initialize_global_comm' which is not static
   pthread_mutex_unlock(&zoltan_global_mpi_lock);
                         ^~~~~~~~~~~~~~~~~~~~~~
/ascldap/users/jhu/trilinos/Trilinos-trilinos-for-sake/packages/zoltan/src/Utilities/Communication/comm_default.c:61:3: warning: 'Zoltan_Global_MPI_Comm' is static but used in inline function 'zoltan_initialize_global_comm' which is not static
   Zoltan_Global_MPI_Comm = comm;
   ^~~~~~~~~~~~~~~~~~~~~~
/ascldap/users/jhu/trilinos/Trilinos-trilinos-for-sake/packages/zoltan/src/Utilities/Communication/comm_default.c:60:23: warning: 'zoltan_global_mpi_lock' is static but used in inline function 'zoltan_initialize_global_comm' which is not static
   pthread_mutex_lock(&zoltan_global_mpi_lock);
                       ^~~~~~~~~~~~~~~~~~~~~~
```

Could these be related to the recent work to eliminate `MPI_COMM_WORLD`?

@trilinos/zoltan 
@JacobDomagala

--- a/packages/zoltan/src/Utilities/Communication/comm_default.c
+++ b/packages/zoltan/src/Utilities/Communication/comm_default.c
@@ -1,4 +1,4 @@
-/* 
+/*
  * @HEADER
  *
  * ***********************************************************************
@@ -51,19 +51,18 @@
 /* if C++, define the rest of this header file as extern C */
 extern "C" {
 #endif
-
 static pthread_mutex_t zoltan_global_mpi_lock;
 static MPI_Comm Zoltan_Global_MPI_Comm = MPI_COMM_WORLD; // CHECK: ALLOW MPI_COMM_WORLD
 
 /* Function to set the default communicator */
-inline void zoltan_initialize_global_comm(MPI_Comm comm) {
+void zoltan_initialize_global_comm(MPI_Comm comm) {
   pthread_mutex_lock(&zoltan_global_mpi_lock);
   Zoltan_Global_MPI_Comm = comm;
   pthread_mutex_unlock(&zoltan_global_mpi_lock);
 }
 
 /* Function to get the default communicator */
-inline MPI_Comm zoltan_get_global_comm() {
+MPI_Comm zoltan_get_global_comm() {
   pthread_mutex_lock(&zoltan_global_mpi_lock);
   MPI_Comm comm = Zoltan_Global_MPI_Comm;
   pthread_mutex_unlock(&zoltan_global_mpi_lock);

